src/Mod/Sketcher/Gui/ViewProviderSketch.cpp

```diff
@@ -3899,4 +3816,153 @@ bool ViewProviderSketch::isInEditMode() const
+                    else {
+                        ++selectedConics;
+                    }
+                }
+            }
+            else if (name.substr(0, 4) == "Vert") {
+                ++selectedPoints;
+            }
+            else if (name.substr(0, 4) == "Cons") {
+                ++selectedConstraints;
+            }
+        }
+
+        if (selectedEdges >= 1 && selectedPoints == 0) {
+            // Here you want to make special cases for line vs arcs vs circle and so on.
+            geom << "Sketcher_Dimension";
+            if (selectedConics == 0) {
+                geom << "Sketcher_ConstrainHorVer"
+                << "Sketcher_ConstrainVertical"
+                << "Sketcher_ConstrainHorizontal";
+            }
+            else if (selectedConics > 1 && selectedLines == 0){
+                geom << "Sketcher_ConstrainCoincidentUnified"
+                    << "Sketcher_ConstrainTangent"
+                    << "Sketcher_ConstrainEqual";
+            }
+            else if (selectedConics == 1 && selectedLines == 1){
+                geom << "Sketcher_ConstrainTangent";
+            }
+            if (selectedLines > 1 && selectedConics == 0) {
+                geom << "Sketcher_ConstrainParallel"
+                    << "Sketcher_ConstrainPerpendicular"
+                    << "Sketcher_ConstrainTangent"
+                    << "Sketcher_ConstrainEqual";
+            }
+        }
+        else if (selectedEdges == 1 && selectedPoints >= 1) {
+            // Here you want to make special cases for line vs arcs vs circle and so on.
+            geom << "Sketcher_Dimension";
+            if (selectedConics == 0) {
+                geom << "Sketcher_ConstrainCoincidentUnified"
+                    << "Sketcher_ConstrainHorVer"
+                    << "Sketcher_ConstrainVertical"
+                    << "Sketcher_ConstrainHorizontal";
+                if (selectedPoints == 2) {
+                geom << "Sketcher_ConstrainSymmetric";
+                }
+            }
+            else {
+                geom << "Sketcher_ConstrainCoincidentUnified"
+                    << "Sketcher_ConstrainPerpendicular"
+                    << "Sketcher_ConstrainTangent";
+            }
+        }
+        else if (selectedEdges == 0 && selectedPoints >= 1) {
+                geom << "Sketcher_Dimension";
+
+                if (selectedPoints > 1) {
+                geom << "Sketcher_ConstrainCoincidentUnified"
+                    << "Sketcher_ConstrainHorVer"
+                    << "Sketcher_ConstrainVertical"
+                    << "Sketcher_ConstrainHorizontal";
+                }
+        }
+        else if (selectedConstraints >= 1) {
+            geom << "Sketcher_ToggleDrivingConstraint"
+                << "Sketcher_ToggleActiveConstraint"
+                << "Sketcher_SelectElementsAssociatedWithConstraints"
+                << "Separator"
+                << "Std_Delete";
+                goto menucreation;
```

Comment:
goto is considered bad practice - try to write the code without it. In that particular case it is harmless but we should not introduce it without a good reason. 

Here you could extract code that adds items to `geom` to another function like `fillMenuEntries(geom)` and then you could simply replace the goto with return statement.

Reply:
Or alternatively you can add a boolean 'addRestOfMenu = true' and set it to false when its constraints selected.

Reply:
I've now done another if statement: if no constraints are in the selection, only then add the rest. Hope this is ok too?

Reply:
No because the other cases ignore constraints. So perhaps you have 2 lines and 1 constraint, in which case the constraint is ignored. So instead of testing if there are no constraints, you need to test if all the rest is empty (edges and points both empty).